### PR TITLE
fixed building errors

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

This should get ISIS building again. Compile error in newhorizons tests and test CMake file globbing always failed since libraries don't exist at configure time.  

## Related Issue

None, Jenkins is failing to build.

## Motivation and Context

We want things to build 

## How Has This Been Tested?

Builds and Functional tests passing locally 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/CMakeLists.txt [hunk 1]
```diff
@@ -8,11 +8,41 @@ file(GLOB test_source "${CMAKE_SOURCE_DIR}/tests/*.cpp")
 file(GLOB isis_libs "${CMAKE_BINARY_DIR}/lib/*.so")
 file(GLOB isis_mac_libs "${CMAKE_BINARY_DIR}/lib/*.dylib")
 
+set(MISSION_LIBS
```
Comment (krlberry) [inline isis/tests/CMakeLists.txt:11]: :+1: Huge improvement, thank you. Will need to remember to update this list if a new mission gets added unless this is updated again in the future to get a list of the current missions.

File: isis/tests/CMakeLists.txt [hunk 2]
```diff
@@ -8,11 +8,41 @@ file(GLOB test_source "${CMAKE_SOURCE_DIR}/tests/*.cpp")
...
+  mgs
+  mro
+  near
+  newhorizons
+  odyssey
+  osirisrex
+  rosetta
+  tgo
+  viking
+  voyager)
```
Comment (Kelvinrr) [inline isis/tests/CMakeLists.txt:38]: Propagating these programmatically looked like a pain without global properties, so I listed them manually for now.

File: isis/tests/FunctionalTestsLeisa2isis.cpp [hunk 3]
```diff
@@ -63,6 +63,7 @@ TEST(leisa2isisTest, leisa2isisTestDefault) {
 
   // BandBin Group
   // Check size, first, 2 middle, and last values? Enough?
+  PvlGroup &bandbin = isisLabel->findGroup("BandBin", Pvl::Traverse);
```
Comment (krlberry) [inline isis/tests/FunctionalTestsLeisa2isis.cpp:66]: I must have accidentally deleted this when doing comment-removal cleanup. Thank you for catching and fixing this!

**Review by krlberry — APPROVED** [review]

**Review by jessemapel — APPROVED** [review]
:+1: for not globbing

File: isis/tests/CMakeLists.txt (hunk 2 above)
Comment (krlberry) [reply isis/tests/CMakeLists.txt:38]: Agreed! That's why I didn't make the 'correct' change in the first place.

**Comment by Kelvinrr** [comment]
@kberryUSGS @jessemapel The Pr is building but some random tests are failing. Looking into them now, everyone OK with merging this and addressing tests in separate PR?

**Comment by jessemapel** [comment]
@Kelvinrr Yeah I'm good with that

**Pull request merged**